### PR TITLE
The classes where the tests were done

### PR DESCRIPTION
**Pull request by morojenkAM**
I made the first tests using mockito and jUnit

### CODE REVIEW

File: src/main/java/ro/developmentfactory/students/Main.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
+package ro.developmentfactory.students;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.*;
+
+public class Main {
```
Comment (olga-casian) [inline src/main/java/ro/developmentfactory/students/Main.java:5]: the code in this class is not really testable. I would recommend you to:
* write a `StudentService` class that will have a separate method for each operation that is in this class. examples of its methods are `getByFirstNameStratingWithA(List <Student> students)`, `getTotalStudensCount(List <Student> students)`, `getStudentMaxAge(List <Student> students)`, ...
* you can either instantiate `StudentService` by passing a `List <Student>` in a constructor or you can pass `List <Student>` to every method. if you pass `List <Student>` in methods, you will have more flexibility in testing
* then you can still use this `Main` class to call each service method for a quick test

File: src/test/java/ro/developmentfactory/students/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+                new Student("Ann", "Gurnmeister", 56, "Female", "Biotech Engineering", 64),
+                new Student("Elon", "Gated", 27, "Male", "Mechanical Engineering", 90),
+                new Student("Justin", "Case", 26, "Male", "Computer Engineering", 340),
+                new Student("Dianne", "Ameter", 31, "Female", "Biotech Engineering", 128),
+                new Student("Joss", "Sticks", 23, "Male", "Computer Engineering", 20)
+        );
+    }
+
+    @Test
+    void FilteredStudentsTest() {
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/students/MainTest.java]: please read on naming conventions for test methods, [this is an example](https://medium.com/@stefanovskyi/unit-test-naming-conventions-dd9208eadbea). we usually prefer something like "test <here is your method name> should find right matches" or "test <here is your method name> should return empty list for empty students" or "test <here is your method name> should return empty list when students is null" - depends what you test. naming can be more generic if you are using parametrized tests. you can also use `@DisplayName("more readable text here")`

File: src/test/java/ro/developmentfactory/students/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+        List<Student> filteredStudents = students.stream()
+                .filter(a -> a.getFirstName().startsWith("A"))//Find a list of students whose first name starts with 'A'
+                .collect(toList());
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/students/MainTest.java]: this part contains actual implementation which you should never put in the test. instead, you should create an instance of `StudentService` and call the method of this instance that is your test subject. it will be easier for you to start writing good tests if you add commented-out lines with Given, When, Then to your tests. ideally, you want to test all relevant usages of your service - those usually come with the requirement but not only. you can think about how others could use your code. if you extract code to `StudentService`, in theory `List<Student> students` can be null, empty list, a list that will not have any matches for some of your methods, and a list that will have 1 or more matches. usually, we want to cover ~80% of the most important test cases but because you are learning and practicing now, you can cover them all. it's much faster and easier to do it with parametrized tests

File: src/test/java/ro/developmentfactory/students/MainTest.java (hunk 2 above)
Comment (olga-casian) [reply src/test/java/ro/developmentfactory/students/MainTest.java]: this applies to all tests in this class

File: src/test/java/ro/developmentfactory/students/MainTest.java (hunk 3 above)
Comment (olga-casian) [reply src/test/java/ro/developmentfactory/students/MainTest.java]: this applies to all tests in this class

File: src/test/java/ro/developmentfactory/arrayProblem/IntermediateSumaTest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
+package ro.developmentfactory.arrayProblem;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+class IntermediateSumaTest {
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/arrayProblem/IntermediateSumaTest.java]: this test class is well done!

File: src/test/java/ro/developmentfactory/arrayProblem/MainTest.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    void setUp(){
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @AfterEach
+    void tearDown(){
+    }
+
+    @Test
+    void testExecutorServiceCreation(){
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/arrayProblem/MainTest.java]: you don't need to test this - it is tested already in java language specification. in our tests, we are assuming that we only test the code that we write. then you need to ask yourself about how it should behave and what it should produce

File: src/test/java/ro/developmentfactory/arrayProblem/MainTest.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        ExecutorService executor = Executors.newFixedThreadPool(10);
+        assertNotNull(executor);
+        assertTrue(executor instanceof ThreadPoolExecutor);
+        assertEquals(10, ((ThreadPoolExecutor) executor).getCorePoolSize());
+        executor.shutdown();
+    }
+
+    @Test
+    void testArrayChunking(){
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/arrayProblem/MainTest.java]: we should not care about the "internals" of your implementation to this level. even if the task was:

> Write a program that will accept a large array of integers as input and calculate the sum of all integers in 5 threads, then add the result together, returning the sum.

it's quite hard to check this without assuming how it works under the hood (`array.length / numberOfTasks`) and in tests we should never assume that we know these details. in this case, we are more interested that by the end the right sum is calculated using your interfaces. you might want to do an extra step to have your code more testable: either write `CinemaService` that will accept either `OptimisticLocking` or `PessimisticLocking` in its constructor, `CinemaService` will have then only 1 method: `bookSeats(int numSeats)`; or write `CinemaService` with 2 methods: `bookSeatsOptimistic(int numSeats)` and `bookSeatspessimistic(int numSeats)` and `CinemaService` should accept both `OptimisticLocking` and `PessimisticLocking` in its constructor. the 1st option will produce a more clean code in this case

File: src/test/java/ro/developmentfactory/arrayProblem/MainTest.java [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        int[] array = { 23, 45, 67, 89, 12, 34, 56, 78, 90, 1 };
+        int numberOfTasks = 5;
+
+        int chunkSize = array.length / numberOfTasks;
+        assertEquals(2,chunkSize);
+    }
+
+    @Test
+    void testExecutorShutdown() throws InterruptedException {
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/arrayProblem/MainTest.java]: same comment: you don't need to have those assumptions about internals

File: src/test/java/ro/developmentfactory/arrayProblem/MainTest.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        }
+
+        executor.shutdown();
+        assertTrue(executor.awaitTermination(10,TimeUnit.SECONDS));
+
+        assertTrue(executor.isTerminated());
+    }
+
+    @Test
+    void testTotalSumCalculation() throws InterruptedException {
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/arrayProblem/MainTest.java]: you should never copy-paste your code into the tests, rather do the following for your test cases:

1. **Given**: prepare the initial state of the system that you want to test (in this case you need to make sure that you use `CinemaService` with the right locking - optimistic or pessimistic - and ideally you want to know what's the cinema capacity, so you can pass 100 in the constructor)
2. **When**: call your method (literally `cinemaService.bookSeats(X)`)
3. **Then**: check that the result is what you actually expect to be (was it an exception? was it the right exception class? what `cinemaService.bookSeats(X)` returned? - you might return true if the booking was successful, you might have a getter for the amount of seats available to check after booking that it was decreased to the right number)

File: src/main/java/ro/developmentfactory/students/StudentService.java [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+        return students.stream().filter(a -> a.getFirstName().startsWith("A")).collect(toList());
+    }
+
+    public Map<String, Set<String>> getNamesByDepertament(List<Student> students) {
+        return students.stream().collect(groupingBy(Student::getDepartamentName, mapping(Student::getFirstName, toSet())));
+    }
+
+    public String getTotalStudents(List<Student> students) {
+        long total = students.stream().count();
+        return "\n Total number of students: " + total;
```
Comment (olga-casian) [inline src/main/java/ro/developmentfactory/students/StudentService.java]: this method should return an integer, not a `String`. it can be `OptionalInt`, `Optional<Integer>`, `Integer`, `int` - the best is `OptionalInt`

File: src/test/java/ro/developmentfactory/students/StudentServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,406 @@
...
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class StudentServiceTest {
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/students/StudentServiceTest.java:14]: you are not testing your service in this test class

File: src/test/java/ro/developmentfactory/arrayProblem/MainTest.java [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+class MainTest {
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/arrayProblem/MainTest.java]: this class does not test your code, please see old comments

File: src/test/java/ro/developmentfactory/arrayProblem/MainTest.java [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+
+    @AfterEach
+    void tearDown() {
+    }
+
+    @Test
+    @DisplayName("Test array chunking")
+    void testArrayChunking_ValidArray_DividedChunks() {
+        int[] array = { 23, 45, 67, 89, 12, 34, 56, 78, 90, 1 };
+        int numberOfTasks = 5;
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/arrayProblem/MainTest.java]: this code is not used

File: src/test/java/ro/developmentfactory/arrayProblem/MainTest.java [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+
+        int chunkSize = 2;
+        assertEquals(2, chunkSize);
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/arrayProblem/MainTest.java]: this test tests assignment of 2 to `int` - not what you want to test

File: src/test/java/ro/developmentfactory/Cinema/CinemaServiceTest.java [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+    public void setup() {
+        optimisticLocking = new OptimisticLocking();
+        pessimisticLocking = new PessimisticLocking();
+        cinemaService = new CinemaService(optimisticLocking, pessimisticLocking);
+    }
+
+    @Test
+    public void testBookSeatsOptimistic() throws NotEnoughSeatsException, SeatsAlreadyReservedException {
+        // Given
+        int initialAvailableSeats = 100;
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/Cinema/CinemaServiceTest.java]: this variable is never used in this and all other tests in this class

File: src/test/java/ro/developmentfactory/Cinema/NotEnoughSeatsExceptionTest.java [hunk 15]
```diff
@@ -0,0 +1,56 @@
+package ro.developmentfactory.Cinema;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class NotEnoughSeatsExceptionTest {
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/Cinema/NotEnoughSeatsExceptionTest.java]: you don't need to write separate tests for your custom exception only, it's covered in `CinemaServiceTest` class

File: src/test/java/ro/developmentfactory/Cinema/SeatsAlreadyReservedExceptionTest.java [hunk 16]
```diff
@@ -0,0 +1,55 @@
+package ro.developmentfactory.Cinema;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class SeatsAlreadyReservedExceptionTest {
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/Cinema/SeatsAlreadyReservedExceptionTest.java]: you don't need to write separate tests for your custom exception only, it's covered in `CinemaServiceTest` class

File: src/test/java/ro/developmentfactory/Cinema/MainTest.java [hunk 17]
```diff
@@ -0,0 +1,64 @@
+package ro.developmentfactory.Cinema;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class Main {
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/Cinema/MainTest.java]: you don't need this test class now as it's covered in `CinemaServiceTest` class

File: src/main/java/ro/developmentfactory/Cinema/CinemaService.java [hunk 18]
```diff
@@ -11,13 +11,15 @@ public CinemaService(OptimisticLocking optimisticLocking, PessimisticLocking pes
...
+    public boolean bookSeatsOptimistic(int numSeats) throws NotEnoughSeatsException, SeatsAlreadyReservedException {
         optimisticLocking.bookSeats(numSeats);
+        return false;
     }
 
     // Method to book seats using pessimistic locking
-    public void bookSeatsPessimistic(int numSeats) throws NotEnoughSeatsException, SeatsAlreadyReservedException {
+    public boolean bookSeatsPessimistic(int numSeats) throws NotEnoughSeatsException, SeatsAlreadyReservedException {
         pessimisticLocking.bookSeats(numSeats);
+        return false;
```
Comment (olga-casian) [inline src/main/java/ro/developmentfactory/Cinema/CinemaService.java]: that would make sense for the locking part to return the result of the operation. returning `false` here is not tied to any logic or result and is confusing

File: src/test/java/ro/developmentfactory/Cinema/CinemaServiceTest.java [hunk 19]
```diff
@@ -2,86 +2,88 @@
...
-        int initialAvailableSeats = 100;
-        cinemaService = new CinemaService(new OptimisticLocking(), new PessimisticLocking());
+        int numSeatsToBook = 150;
+
+        // When, Then
+        assertThrows(NotEnoughSeatsException.class, () -> cinemaService.bookSeatsOptimistic(numSeatsToBook));
+    }
+
+    @Test
+    void testBookSeatsOptimistic_SeatsAlreadyReserved() throws NotEnoughSeatsException, SeatsAlreadyReservedException {
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/Cinema/CinemaServiceTest.java]: this one does not actually test the case when seats were already reserved

File: src/test/java/ro/developmentfactory/Cinema/CinemaServiceTest.java [hunk 20]
```diff
@@ -2,86 +2,88 @@
...
-        // Then
-        assertThrows(NotEnoughSeatsException.class, () -> cinemaService.bookSeatsOptimistic(101), "Exception should be thrown when booking more seats than available (optimistic).");
-        assertThrows(NotEnoughSeatsException.class, () -> cinemaService.bookSeatsPessimistic(101), "Exception should be thrown when booking more seats than available (pessimistic).");
+        // When, Then
+        assertThrows(NotEnoughSeatsException.class, () -> cinemaService.bookSeatsPessimistic(numSeatsToBook));
     }
 
     @Test
-    public void testBookSeatsThrowsSeatsAlreadyReservedException() throws NotEnoughSeatsException, SeatsAlreadyReservedException {
+    void testBookSeatsPessimistic_SeatsAlreadyReserved() throws NotEnoughSeatsException, SeatsAlreadyReservedException {
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/Cinema/CinemaServiceTest.java]: this one does not actually test the case when seats were already reserved

File: src/test/java/ro/developmentfactory/Cinema/CinemaServiceTest.java [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import ro.developmentfactory.Cinema.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class CinemaServiceTest {
+
+    private CinemaService cinemaService;
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/Cinema/CinemaServiceTest.java:11]: what will happen if you pass a negative number or 0 to `bookSeatsPessimistic` or `bookSeatsOptimistic`? they are valid cases too

File: src/test/java/ro/developmentfactory/arrayProblem/MainTest.java [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @AfterEach
+    void tearDown() {
+    }
+
+    @Test
+    @DisplayName("Test total sum calculation")
+    void testTotalSumCalculation_ExecutedTasks_CorrectTotalSum() throws InterruptedException {
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/arrayProblem/MainTest.java]: you should never copy-paste your code into the tests, rather apply the same approach as in `CinemaService`:

- write a service, ex: `ArraySumService`
- put all the logic that manages `IntermediateSuma` there into a method with a signature ex: `public int calculate(int[] array)`
- create an instance of `ArraySumService` in tests and test the invocations of `calculate()`. it makes sense to test for an empty list, a list with 1 element of 0, a list with negative numbers, positive numbers, mixed numbers, and, ideally, a very-very large list. you already covered the majority in `IntermediateSumaTest`, but because you assume your clients to use not `IntermediateSuma` but rather `ArraySumService`, in real life that would probably make sense to test `ArraySumService`

File: src/test/java/ro/developmentfactory/students/StudentServiceTest.java [hunk 23]
```diff
@@ -0,0 +1,251 @@
...
+                new Student("Ann", "Gurnmeister", 56, "Female", "Biotech Engineering", 64),
+                new Student("Elon", "Gated", 27, "Male", "Mechanical Engineering", 90),
+                new Student("Justin", "Case", 26, "Male", "Computer Engineering", 340),
+                new Student("Dianne", "Ameter", 31, "Female", "Biotech Engineering", 128),
+                new Student("Joss", "Sticks", 23, "Male", "Computer Engineering", 20)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideFirstNamePrefixes")
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/students/StudentServiceTest.java]: you are not using a parametrized test here, so it can be removed. the method `provideFirstNamePrefixes()` too

File: src/test/java/ro/developmentfactory/students/StudentServiceTest.java [hunk 24]
```diff
@@ -0,0 +1,251 @@
...
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import java.util.*;
+import static org.junit.jupiter.api.Assertions.*;
+import static java.util.stream.Collectors.*;
+
+public class StudentServiceTest {
+
+    private StudentService studentService;
+    private List<Student> students;
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/students/StudentServiceTest.java]: you don't need to initialize `students` in `@BeforeEach` as it will run as many times as you have tests and your list of students is not supposed to be changed anyway. so you can initialize it here and mark it `final`

File: src/test/java/ro/developmentfactory/students/StudentServiceTest.java [hunk 25]
```diff
@@ -0,0 +1,251 @@
...
+    void GroupNamesByDepartment_EmptyStudentList() {
+        // Simulate an empty list of students
+        List<Student> emptyStudentsList = Collections.emptyList();
+        Map<String, Set<String>> namesByDepertament = emptyStudentsList.stream()
+                .collect(groupingBy(
+                        Student::getDepartamentName,
+                        mapping(
+                                Student::getFirstName,
+                                toSet())
+                ));
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/students/StudentServiceTest.java]: you should rather call `studentService.getNamesByDepertament(emptyStudentsList)` instead

File: src/test/java/ro/developmentfactory/students/StudentServiceTest.java [hunk 26]
```diff
@@ -0,0 +1,251 @@
...
+        assertTrue(computerEngineers.contains("Joss"));
+
+        Set<String> biotechEngineers = namesByDepertament.get("Biotech Engineering");
+        assertTrue(biotechEngineers.contains("Ann"));
+        assertTrue(biotechEngineers.contains("Dianne"));
+    }
+
+    @Test
+    @DisplayName("Group names by department with empty student list")
+    void GroupNamesByDepartment_EmptyStudentList() {
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/students/StudentServiceTest.java]: if you want another edge case - you could have a list with `null` element. you are not testing any edge cases for the rest of the methods, just be aware that normally for production code you should have it

File: src/main/java/ro/developmentfactory/Cinema/OptimisticLocking.java [hunk 27]
```diff
@@ -19,7 +19,8 @@ public void bookSeats(int numSeats) throws NotEnoughSeatsException,SeatsAlreadyR
         while(!occupiedSeats.compareAndSet(currentOccupiedSeats, currentOccupiedSeats + numSeats)) {
             currentOccupiedSeats = occupiedSeats.get();
         }
-            }
+        return false;
```
Comment (olga-casian) [inline src/main/java/ro/developmentfactory/Cinema/OptimisticLocking.java]: usually, we return true when something succeeds and false when failed, so it would be nice to return true

Comment (olga-casian) [reply src/main/java/ro/developmentfactory/Cinema/OptimisticLocking.java]: `PessimisticLocking.bookSeats()` still is void

File: src/test/java/ro/developmentfactory/Cinema/CinemaServiceTest.java [hunk 28]
```diff
@@ -0,0 +1,96 @@
...
+        boolean firstBookingResult = cinemaService.bookSeatsPessimistic(numSeats);
+        assertTrue(firstBookingResult); // Assume seats are available and booking succeeds
+
+        // Second booking attempt (should throw SeatsAlreadyReservedException or return false)
+        boolean secondBookingResult = cinemaService.bookSeatsPessimistic(numSeats);
+        assertTrue(secondBookingResult); // Expecting false because seats are already reserved
+    }
+
+    @Test
+   void testBookSeatsOptimistic_SeatsAlreadyReserved() throws SeatsAlreadyReservedException, NotEnoughSeatsException {
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/Cinema/CinemaServiceTest.java:70]: from the method name and the fact that there are no other tests for the exception `SeatsAlreadyReservedException`, I'd assume that this test should check if `SeatsAlreadyReservedException` exception was thrown (like in `testBookSeatsOptimistic_NotEnoughSeatsException` test).

and what is the purpose of this test? to me it looks like it tests what `testBookSeatsOptimistic_SuccessfulBooking` test does but less

File: src/test/java/ro/developmentfactory/Cinema/CinemaServiceTest.java [hunk 29]
```diff
@@ -0,0 +1,96 @@
...
+
+        // When
+        cinemaService.bookSeatsPessimistic(numSeatsToBook);
+
+        // Then
+        assertEquals(initialAvailableSeats - numSeatsToBook, cinemaService.getAvailableSeatsPessimistic());
+    }
+
+    @Test
+   void testBookSeatsPessimistic_SeatsAlreadyReserved() throws SeatsAlreadyReservedException, NotEnoughSeatsException {
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/Cinema/CinemaServiceTest.java:56]: same as for `testBookSeatsOptimistic_SeatsAlreadyReserved`

File: src/test/java/ro/developmentfactory/arrayProblem/ArraySumServiceTest.java [hunk 30]
```diff
@@ -0,0 +1,70 @@
...
+    @Test
+    public void testCalculate_MixedNumbers() {
+        ArraySumService arraySumService = new ArraySumService(4);
+        int[] array = {-1, 2, 0, 5, -3};
+        int result = arraySumService.calculate(array);
+        assertEquals(3, result);
+    }
+
+    @Test
+    public void testCalculate_VeryLargeArray() {
```
Comment (olga-casian) [inline src/test/java/ro/developmentfactory/arrayProblem/ArraySumServiceTest.java:56]: by saying that you should test on a very large array, I meant that the array should have many-many elements, not 1 large number :) if you want to challenge yourself, you can even write array's elements to a file that would not fit into your memory and see how you can approach it differently to read the values and calculate the sum. but it's fine to leave it like this so far
